src/data/actions/codes.js

```diff
@@ -0,0 +1,53 @@
+import {
+  REQUEST_CODES_REQUEST,
+  REQUEST_CODES_SUCCESS,
+  REQUEST_CODES_FAILURE,
+  CLEAR_REQUEST_CODES,
+} from '../constants/codes';
+
+import LmsApiService from '../services/LmsApiService';
+
+const requestCodesRequest = () => ({
+  type: REQUEST_CODES_REQUEST,
+});
+
+const requestCodesSuccess = success => ({
+  type: REQUEST_CODES_SUCCESS,
+  payload: {
+    success,
+  },
+});
+
+const requestCodesFailure = error => ({
+  type: REQUEST_CODES_FAILURE,
+  payload: {
+    error,
+  },
+});
+
+const clearRequestCodesEvent = () => ({ type: CLEAR_REQUEST_CODES });
+
+const requestCodes = (enterpriseName, emailAddress, numberOfCodes) => (
```

Comment:
Related to the other comments, this function could accept just a single object:

```javascript
const requestCodes = (options) => (
  (dispatch) => {
    dispatch(requestCodesRequest());
    return LmsApiService.requestCodes(options)
      .then(() => { ... })
      .catch(() => {... });
  }
);
```
The `LmsApiService.requestCodes()` function would extract the values out using object destructuring.